### PR TITLE
Add test with restart/down container

### PR DESCRIPTION
**Pull request by mdonaka**
# Summary
Add test with restart/down container.

# Purpose
Automation of abnormal test

# Contents
- Add container_test
  - container up test
  - Various request test with container down
  - Various request test with container restart

# Testing Methods Performed
- make test t=container_test p=medium
- CI

### CODE REVIEW

File: scripts/container_test/medium_test.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# build処理を記述する関数
+build() {
+    docker buildx bake $COMPOSE_FILES_OPT medium_container dev_bts dev_cc1 dev_cc2 dev_cc3 dev_mc1 dev_mc2 dev_mc3 --load
+}
+
+# runの前に実行されるsetup処理を記述する関数
+# INFO: runを冪等に実行できるように初期化を行う
+setup() {
+    rm /tmp/QuickMPC
+    docker-compose $COMPOSE_FILES_OPT down -v
```
Comment (come2ry) [inline scripts/container_test/medium_test.sh:13]:
```suggestion
    rm /tmp/QuickMPC && docker-compose $COMPOSE_FILES_OPT down -v
```

File: scripts/container_test/medium_test.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+# run処理を記述する関数
+# NOTE: この関数は例外的にワンライナーで書かなくて良い
+run() {
+    # DooDでmountしてコンテナを立てるようとするとpathがデーモンのあるhostから参照されてしまう
+    # コンテナ内からhostのpathが見えず指定できないため，明示的な場所にQuickMPCを作ってここを指定するようにする
+    ln -s $(pwd)/../ /tmp/QuickMPC
+    docker-compose $COMPOSE_FILES_OPT up medium_container
```
Comment (come2ry) [inline scripts/container_test/medium_test.sh:22]:
```suggestion
    ln -s $(pwd)/../ /tmp/QuickMPC && docker-compose $COMPOSE_FILES_OPT up medium_container
```

**Review by come2ry — CHANGES_REQUESTED** [review]

**Review by arukuka — APPROVED** [review]
Thank you for your contribution!

File: scripts/container_test/medium_test.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+# runの後に実行されるteardown処理を記述する関数
+# INFO: runの後に副作用を残さないように初期化を行う
+teardown() {
+    rm /tmp/QuickMPC
+    docker-compose $COMPOSE_FILES_OPT down -v
```
[the remark this answers is not in the document]
Comment (mdonaka) [reply scripts/container_test/medium_test.sh]: Thanks for pointing that out!
Fixed (and added option `-f`) with 48d13a1ad4df2afbb8134fcbe01d4eb34718d119.

**Comment by mdonaka** [comment]
It turns out that the test fails once every few dozen times.
This will probably be solved by introducing a retry policy in X.
Therefore, I would set the PR to "draft" until this is implemented.

This PR itself remains unchanged.

**Comment by come2ry** [comment]
@mdonaka 
I think the following issues with this PR.
-Using DooD, we have to create `scripts/docker-compose.unit.yml`, which is not DRY and has a large maintenance cost.

I believe this issue is due to the difference in paths between host and medium_container due to DooD, which I think can be solved by stopping DooD and running the test code directly from host.

Is there something wrong with running tests directly from host?
If there is no problem, the `scripts/docker-compose.unit.yml`, `medium_test_container` and `scripts/container_test/Dockerfile` can be deleted, making it easier to maintain and understand The code will be easier to maintain and understand.

**Comment by mdonaka** [comment]
@come2ry 
Thank you for your comment!
I had created a container based on the stereotype, but it seems that there is no problem in the form of direct execution from the host!
I will revise it in the form you suggested. Thank you very much!

**Comment by mdonaka** [comment]
I opened the PR because we did not receive any notification if the PR was a draft.

**Comment by mdonaka** [comment]
@come2ry 
I changed to run tests directly. Please review again!

**Review by come2ry — APPROVED** [review]
LGTM！

**Pull request merged**